sale_order_product_recommendation/wizards/sale_order_recommendation.py

```diff
@@ -215,3 +231,21 @@ def _prepare_new_so_line(self, line_form, sequence):
         line_form.product_id = self.product_id
         line_form.sequence = sequence
         line_form.product_uom_qty = self.units_included
+
+    def _get_last_sale_price_product(self):
+        """
+        Get last price from last order.
+        Use sudo to read sale order from other users like as other commercials.
+        """
+        self.ensure_one()
+        so = self.env["sale.order"].sudo().search([
+            ("partner_id", "=", fields.first(self).partner_id.id),
+            ("confirmation_date", "!=", False),
+            ("state", "not in", ('draft', 'sent', 'cancel')),
+            ("order_line.product_id", "=", self.product_id.id)
```

Comment:
Optimal performance because order_line field uses auto_join:
https://github.com/odoo/odoo/blob/8c54ed42eef16817f198f10697820eddcc7d0c74/addons/sale/models/sale.py#L174